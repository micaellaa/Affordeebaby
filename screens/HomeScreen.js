import { useNavigation } from '@react-navigation/core'
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { signOut } from 'firebase/auth';
import React, { useState } from 'react';
import { KeyboardAvoidingView, StyleSheet, Text, TextInput, View, Image, Alert } from 'react-native';
import { ImageBackground, TouchableHighlight, TouchableOpacity } from 'react-native-web';
import { authentication } from "../firebase/firebase-config";
import COLORS from '../consts/colors';
import Icon from 'react-native-vector-icons/MaterialIcons';
import background from "../assets/palmshadow-bg.png";


const HomeScreen = () => {
  
  const navigation = useNavigation()

  const handleSignOut = () => {
    signOut(authentication)
    .then(() => {
      navigation.replace("Login")
    })
    .catch(error => alert(error.message))
  }

  const shopByRetailer = () => {
    navigation.navigate("Retailers")
  }

  const quickShop = () => {
    navigation.navigate("Quickshop")
  }

  /*const manageOrders = () => {
    navigation.navigate("Quickshop")

  const discounts = () => {
    navigation.navigate("Discounts")
  }*/

  const menuOptions = ['Shop', 'Profile', 'Find Friends'];
  const [menuOptionsIndex, setMenuOptionsIndex] = useState(0)

  return (
    <ImageBackground
      source={require('../assets/palmshadow-bg.png')} //stub image
      style={styles.container}
    >

      <View style={styles.menuContainer}> 
        {menuOptions.map((item, index) => (
          <TouchableOpacity 
            key={index}
            activeOpacity={0.8}
            onPress={()=>setMenuOptionsIndex(index)}>
            <Text
            style={[
              styles.menuOptionsText,
              menuOptionsIndex == index && styles.menuOptionsTextSelected,
            ]}>
              {item}
            </Text>
          </TouchableOpacity>
        ))}
      </View>

      <View style={styles.buttonContainer}>
        <TouchableOpacity
          onPress={quickShop}
          style={styles.button}
        >
          <Text style={styles.buttonText}>Quick Shop</Text>
        </TouchableOpacity>
        <TouchableOpacity
          onPress={() => navigation.navigate("Retailers")}
          style={[styles.button, styles.buttonOutline]}
        >
          <Text style={styles.buttonText}>Shop by Retailer</Text>
        </TouchableOpacity>
        <TouchableOpacity
          //onPress={manageOrders}
          style={[styles.button, styles.buttonOutline]} 
        >
          <Text style={styles.buttonText}>Manage My Orders</Text>
        </TouchableOpacity>
        <TouchableOpacity
          //onPress={manageOrders}
          style={[styles.button, styles.buttonOutline]}
        >
          <Text style={styles.buttonText}>Discounts</Text>
        </TouchableOpacity>
      </View>
      
      <View style={styles.container}>
        <Text>Email: {authentication.currentUser?.email}</Text>
        <TouchableOpacity
          onPress={handleSignOut}
          style={styles.button}
        >
          <Text style={styles.buttonText}>Sign out</Text>
        </TouchableOpacity>
      </View>

    </ImageBackground>
  )
}

export default HomeScreen

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
   button: {
    backgroundColor: COLORS.indigo,
    width: '90%',
    padding: 15,
    borderRadius: 10,
    alignItems: 'center',
    marginTop: 40,
  },
  buttonContainer:{
    flex: 1,
    width: '60%',
    justifyContent: 'center',
    alignItems: 'center',
  },
  buttonText: {
    color: 'white',
    fontWeight: '700',
    fontSize: 16,
  },
  menuContainer: {
    flexDirection: 'row',
    width: '80%',
    marginTop: 30,
    marginBottom: 40,
    justifyContent: 'space-between',
  },
  menuOptionsText: {
    fontsize: 16,
    color: 'grey',
    fontWeight: 'bold',
  },
  menuOptionsTextSelected: {
    color: COLORS.indigo,
    paddingBottom: 5,
    borderBottomWidth: 2,
    borderColor: COLORS.indigo,
  },
})