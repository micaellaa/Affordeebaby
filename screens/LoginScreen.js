import { useNavigation } from '@react-navigation/core'
import { KeyboardAvoidingView, StyleSheet, Text, TextInput, View, Image } from 'react-native';
import React, { useState } from 'react';
import { TouchableHighlight, TouchableOpacity } from 'react-native-web';
//import { auth } from '../firebase'
import { authentication } from "../firebase/firebase-config";
import { createUserWithEmailAndPassword, signInWithEmailAndPassword, onAuthStateChanged } from 'firebase/auth';
//import { useEffect } from 'react/cjs/react.production.min';
import { useEffect } from 'react';
import { NavigationContainer } from '@react-navigation/native';
import COLORS from '../consts/colors';

const LoginScreen = () => {
  const [isSignedIn, setIsSignedIn] = useState(false)
  const [email, setEmail] = useState('')
  const [password, setPassword] = useState('') //useState() 
  //const auth = getAuth();

  const navigation = useNavigation()

  //
/*
  const useFirebaseAuthentication = (firebase) => {
    const [authUser, setAuthUser] = useState(null);
    useEffect(() => {
      const unsubscribe = onAuthStateChanged(authentication, 
          authUser => {
            if (authUser) {
              navigation.replace("Home")
            }
            authUser
              ? setAuthUser(authUser)
              : setAuthUser(null);
          },
      );
      return () => {
        unsubscribe();
      }
    }, []);

    return authUser
  }
  */
  

  useEffect(() => {
    return onAuthStateChanged(authentication, user => {
      if (user) {
        navigation.replace("Home") //diff between .replace and .navigate
      }
    })
  }, [])

  /*
  const useFirebaseAuthentication = () => {
    const [authUser, setAuthUser] = useState(null);
    useEffect(() => {
      const unlisten = onAuthStateChanged(authentication, 
        user => {
          if (user) {
            navigation.navigate("Home")
          }
        }
      );
      return () => {
        unlisten();
      }
    }, []);
    return authUser
  }
  */



  // SIGN-UP
  /*const handleSignUp = () => {
    createUserWithEmailAndPassword(authentication, email, password)
      .then(userCredentials => {
        const user = userCredentials.user;
        console.log(user.email);
      })
      .catch(error => alert(error.message))
  }*/

  const handleSignUp = () => {
    navigation.replace("Register")
  }


  // SIGN-IN
  const handleSignIn = () => {
    signInWithEmailAndPassword(authentication, email, password)
      .then(userCredentials => {
        const user = userCredentials.user;
        setIsSignedIn(true);
        console.log('Signed in with: ', user.email);
      })
      .catch(error => alert(error.message))
  }


  return (
    //<Image source={require('../assets/palmshadow-bg.png')}/>
    <KeyboardAvoidingView
      style={styles.container}
      behavior="padding"
    >
      <View style={styles.iconContainer}>  
        <Image 
          style={styles.iconDimensions}
          source={require('../assets/logogradientstroke.png')}/>
      </View>

      <View style={styles.inputContainer}>  
        <TextInput
          placeholder= "Email"
          value={email}
          onChangeText={text => setEmail(text)} //a lambda
          style={styles.input}
        />
        <TextInput
          placeholder= "Password"
          value={password}
          onChangeText={text => setPassword(text)} //why is it set to password
          style={styles.input}
          secureTextEntry
        />
      </View>

      <View style={styles.buttonContainer}>
        <TouchableOpacity
          onPress={handleSignIn}
          style={styles.button}
        >
          <Text style={styles.buttonText}>Login</Text>
        </TouchableOpacity>
        <TouchableOpacity
          onPress={handleSignUp}
          style={[styles.button, styles.buttonOutline]}
        >
          <Text style={styles.buttonOutlineText}>Register</Text>
        </TouchableOpacity>
      </View>

    </KeyboardAvoidingView>
  )
}

export default LoginScreen

const styles = StyleSheet.create({
  iconContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 20,
  },
  iconDimensions: {
    width: 150, 
    height: 150
  },
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  inputContainer:{
    width: '80%'
  },
  input:{
    backgroundColor: "white", //grey
    addingHorizontal: 15,
    paddingVertical: 10,
    borderRadius: 10,
    marginTop: 5,
  },
  button:{
    backgroundColor: COLORS.indigo,
    width: '100%',
    padding: 15,
    borderRadius: 10,
  },
  buttonContainer:{
    width: '60%',
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 40,
  },
  buttonOutline:{
    backgroundColor: "white", //grey
    marginTop: 5,
    borderColor: COLORS.indigo,
    borderwidth: 2,
  },
  buttonText:{
    color: 'white',
    fontWeight: '700',
    fontSize: 16,
  },
  buttonOutlineText:{
    color: COLORS.indigo,
    fontWeight: '700',
    fontSize: 16,
  },
})
