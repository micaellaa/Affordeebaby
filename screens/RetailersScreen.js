import { useNavigation } from '@react-navigation/core'
import { signOut } from 'firebase/auth';
import React from 'react'
import { KeyboardAvoidingView, StyleSheet, Text, TextInput, View } from 'react-native';
import { TouchableOpacity } from 'react-native';
import { authentication } from "../firebase/firebase-config";

const RetailersScreen = () => {
  
  const navigation = useNavigation()

  return (
    <KeyboardAvoidingView
      style={styles.container}
      behavior="padding"
    >

      <View style={styles.buttonContainer}>
        <TouchableOpacity
          //onPress={handleSignIn}
          style={styles.button}
        >
          <Text style={styles.buttonText}>Zara</Text>
        </TouchableOpacity>
        <TouchableOpacity
          onPress={() => navigation.navigate("Quickshop")}
          style={[styles.button, styles.buttonOutline]}
        >
          <Text style={styles.buttonOutlineText}>ASOS</Text>
        </TouchableOpacity>
      </View>
      
    </KeyboardAvoidingView>
  )
}

export default RetailersScreen

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
   button: {
    backgroundColor: '#0782F9',
    width: '60%',
    padding: 15,
    borderRadius: 10,
    alignItems: 'center',
    marginTop: 40,
  },
  buttonContainer:{
    width: '60%',
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 40,
  },
  buttonText: {
    color: 'white',
    fontWeight: '700',
    fontSize: 16,
  },
})