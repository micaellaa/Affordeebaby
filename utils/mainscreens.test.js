import React from "react";
import renderer from "react-test-renderer";
import LoginScreen from "../screens/LoginScreen";
import HomeScreen from "../screens/LoginScreen";
import RetailersScreen from "../screens/RetailersScreen";

jest.mock("assets/logogradientstroke.png");

test("LoginScreen renders correctly", () => {
  const tree = renderer.create(<LoginScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("HomeScreen renders correctly", () => {
  const tree = renderer.create(<HomeScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
