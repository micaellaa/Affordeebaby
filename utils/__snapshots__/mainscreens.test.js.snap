// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": 20,
      }
    }
  >
    <Image
      source={Object {}}
      style={
        Object {
          "height": 150,
          "width": 150,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "width": "80%",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Email"
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 10,
          "marginTop": 5,
          "paddingHorizontal": 15,
          "paddingVertical": 10,
        }
      }
      value=""
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 10,
          "marginTop": 5,
          "paddingHorizontal": 15,
          "paddingVertical": 10,
        }
      }
      value=""
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": 40,
        "width": "60%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#7853cd",
          "borderRadius": 10,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "700",
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "#7853cd",
          "borderRadius": 10,
          "borderwidth": 2,
          "marginTop": 5,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#7853cd",
            "fontSize": 16,
            "fontWeight": "700",
          }
        }
      >
        Register
      </Text>
    </View>
  </View>
</View>
`;

exports[`LoginScreen renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": 20,
      }
    }
  >
    <Image
      source={Object {}}
      style={
        Object {
          "height": 150,
          "width": 150,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "width": "80%",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Email"
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 10,
          "marginTop": 5,
          "paddingHorizontal": 15,
          "paddingVertical": 10,
        }
      }
      value=""
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 10,
          "marginTop": 5,
          "paddingHorizontal": 15,
          "paddingVertical": 10,
        }
      }
      value=""
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": 40,
        "width": "60%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#7853cd",
          "borderRadius": 10,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "700",
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "#7853cd",
          "borderRadius": 10,
          "borderwidth": 2,
          "marginTop": 5,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#7853cd",
            "fontSize": 16,
            "fontWeight": "700",
          }
        }
      >
        Register
      </Text>
    </View>
  </View>
</View>
`;

exports[`RetailerssScreen renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <Image
    source={Object {}}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 30,
        "marginTop": 60,
        "width": "80%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "grey",
              "fontWeight": "bold",
              "fontsize": 16,
            },
            Object {
              "borderBottomWidth": 2,
              "borderColor": "#7853cd",
              "color": "#7853cd",
              "paddingBottom": 5,
            },
          ]
        }
      >
        Shop
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "grey",
              "fontWeight": "bold",
              "fontsize": 16,
            },
            false,
          ]
        }
      >
        Profile
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "grey",
              "fontWeight": "bold",
              "fontsize": 16,
            },
            false,
          ]
        }
      >
        Find Friends
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "width": "80%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={Object {}}
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={Object {}}
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": 40,
        "width": "60%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#0782F9",
          "borderRadius": 10,
          "marginTop": 40,
          "opacity": 1,
          "padding": 15,
          "width": "60%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "700",
          }
        }
      >
        Zara
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#0782F9",
          "borderRadius": 10,
          "marginTop": 40,
          "opacity": 1,
          "padding": 15,
          "width": "60%",
        }
      }
    >
      <Text>
        ASOS
      </Text>
    </View>
  </View>
</View>
`;
